src/main/scala/calc/TypedTree.scala

```diff
@@ -0,0 +1,30 @@
+package calc
+
+import org.scalajs.core.ir.Position
+
+
+sealed abstract class TypedTree {
+  def pos: Position
+  def tp: TreeType
+}
+
+final case class LiteralT(value: Double, tp: TreeType=NoType)
```

Comment:
You should not have default params for `tp`. There should be no reason to create a `TypedTree` without a type, so you want the (Scala) compiler to prevent you from doing so.